java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -208,28 +215,33 @@ public synchronized BarrageTable partialTable(RowSet viewport, BitSet columns, b
         if (subscribed) {
             throw new UncheckedDeephavenException(
                     "BarrageSubscription objects cannot be reused.");
-        } else {
-            // test lock conditions
-            if (resultTable.getUpdateGraph().sharedLock().isHeldByCurrentThread()) {
-                throw new UnsupportedOperationException(
-                        "Cannot create subscription while holding the UpdateGraph shared lock");
-            }
+        }
+        subscribed = true;
 
-            if (resultTable.getUpdateGraph().exclusiveLock().isHeldByCurrentThread()) {
-                completedCondition = resultTable.getUpdateGraph().exclusiveLock().newCondition();
-            }
+        if (!isSnapshot) {
+            resultTable.addSourceToRegistrar();
+            resultTable.addParentReference(this);
+        }
+
+        // test lock conditions
+        if (updateGraph.sharedLock().isHeldByCurrentThread()) {
+            throw new UnsupportedOperationException(
+                    "Cannot create subscription while holding the UpdateGraph shared lock");
+        }
+
+        if (updateGraph.exclusiveLock().isHeldByCurrentThread()) {
+            completedCondition = resultTable.getUpdateGraph().exclusiveLock().newCondition();
+        }
 
-            // Send the initial subscription:
-            observer.onNext(FlightData.newBuilder()
-                    .setAppMetadata(ByteStringAccess.wrap(makeRequestInternal(
-                            viewport, columns, reverseViewport, options, tableHandle.ticketId().bytes())))
-                    .build());
-            subscribed = true;
+        this.viewport = viewport;
+        this.reverseViewport = reverseViewport;
 
+        if (resultTable.isRefreshing()) {
```

Comment:
We've become convinced that we no longer need this listener. The update source needs to handle this, and can do the callback directly.